slitlessutils/core/wfss/config/instrumentconfig.py

```diff
@@ -680,42 +696,45 @@ def from_fitsfile(cls, filename, **kwargs):
         with fits.open(filename, mode='readonly') as hdul:
             h0 = hdul[0].header
 
-            tel = h0['TELESCOP']
-            ins = h0['INSTRUME']
-            if ins == 'WFC3':
-                ins += h0['DETECTOR']
-                disperser = h0['FILTER']
-
-            elif ins == 'ACS':
-                ins += h0['DETECTOR']
-                disperser = h0['FILTER1']
-
-            elif ins == 'NIRISS':
-                disperser = (h0['FILTER'], h0['PUPIL'])
-                kwargs['fwcpos'] = h0['FWCPOS']
+            # check that the OBSTYPE is a spectroscopic image
+            if h0['OBSTYPE'] == 'SPECTROSCOPIC':
+                tel = h0['TELESCOP']
+                ins = h0['INSTRUME']
+                if ins == 'WFC3':
+                    ins += h0['DETECTOR']
+                    disperser = h0['FILTER']
+
+                elif ins == 'ACS':
+                    ins += h0['DETECTOR']
+                    disperser = h0['FILTER1']
+
+                elif ins == 'NIRISS':
+                    disperser = (h0['FILTER'], h0['PUPIL'])
+                    kwargs['fwcpos'] = h0['FWCPOS']
+                else:
+                    raise NotImplementedError(f"Unsupported: {tel = } {ins = }")
+
+                insconf = cls(tel, ins, disperser, **kwargs)
+
+                # update some things because subarrays
+                for detname in insconf.keys():
+                    ext = insconf[detname].extensions['science'].extension
+                    h = fits.getheader(filename, ext)
+
+                    if insconf[detname].naxis[0] != h['NAXIS1']:
+                        insconf[detname].naxis[0] = h['NAXIS1']
+                        insconf.subarray = True
+                    if insconf[detname].naxis[1] != h['NAXIS2']:
```

Comment:
```suggestion
                    elif insconf[detname].naxis[1] != h['NAXIS2']:
```
Seems like this and the next two should be elif instead of if, since they don't need to execute if we've already decided that insconf.subarray should be true.